JPS_Ontology/ontology/ontobuiltenv/OntoBuildingEnvironment.csv

```diff
@@ -118,3 +118,10 @@ representativeFor,Object Property,,,AveragePricePerSqm,PostalCode,,A relationshi
 representativeFor,Object Property,,,AveragePricePerSqm,PostalCode,exactly 1,,https://www.theworldavatar.com/kg/ontobuiltenv/,representative for
 Bank,Class,Non-Domestic,IS-A,,,,A property where customers can deposit and withdraw money.,https://www.theworldavatar.com/kg/ontobuiltenv/,Bank
 Pharmacy,Class,Medical,IS-A,,,,A healthcare facility where medicinal drugs are prepared or sold.,https://www.theworldavatar.com/kg/ontobuiltenv/,Pharmacy
+BuildingFloors,Class,,,Property,,,A concept to describe amount of floors in a building.,https://www.theworldavatar.com/kg/ontobuiltenv/,Building floors
+hasBuildingFloors,Object Property,,,Property,BuildingFloors,exactly 1,A relationship between a property and its amount floors.,https://www.theworldavatar.com/kg/ontobuiltenv/,has building floors
+hasFloorsValue,Data Property,,,BuildingFloors,integer,exactly 1,A relationship between a building floors and its value.,https://www.theworldavatar.com/kg/ontobuiltenv/,has floors value
+hasAccuracy,Data Property,,,BuildingFloors,string,exactly 1,A relationship between a building floors and its accuracy level.,https://www.theworldavatar.com/kg/ontobuiltenv/,has accuracy
```

Comment:
While I understand it's good to mark up the provenance of building floors (i.e., generated from either HDBPropertyInformation.csv, OpenStreetMap, BuildingHeightEstimation). BuildingFloors can't really have accuracy (i.e., high, medium, low) because there is no indication why one dataset is more accurate than the other.

Would something along the line of <BuildingFloors> <derivedFrom> <HDBPropertyInformation/OpenStreetMap/BuildingHeightEstimation> be better than high, medium, low, accuracy?

Reply:
Unsure how that is not easy for further query and identification, can get @YiKaiTsai and @sganguly-cares to weigh in on this

Reply:
@sm453, would appreciate your advice on this.

Reply:
1. It doesn't make sense to store such a long and irregular name of data sources in database. 2. The objective of this concept is to help us know the reliability of data, then we can know which data should be kept. When querying get a data source name, how could the agent know it is a reliable data or not? how to make a decision to update data or not? Even in the future if some new data is coming, how could agent know the data is reliable or not? 

So far, data accuracy level: 1. high: data from reliable departments (e.g. government agencies) 2. mid: open source data without any guarantees (e.g. OSM) 3. low: estimate by height. I think it's very clear and easy for further work.

Reply:
A few comments/suggestions:
1) Using string literals to represent categorical data ("high", "medium", "low") is really the worst way to represent this. It's slow, storage-inefficient, error-prone, and completely non-semantic. I would strongly recommend to change this (pretty much any alternative will be preferable).
2) Using the word "accuracy" leads me to expect some kind of quantified uncertainty, e.g. a percentage error, whereas what you have in mind, I guess, is much less quantitative and more akin to a "level of confidence" that you have in the data (or the entity providing them). I would find it much clearer if you use the latter wording.
3) I agree with @ShinZert's point that rather than associating a subjective assessment of quality (that may offend people!) to a dataset, it would be much preferable to simply store the provenance (where the data came from). Then any "user" can make up their own mind how good the data are, without having to spell it out!
4) If you do decide to go with some kind of categorical solution, I would recommend using subclasses (rather than string datatype properties), because this does not rely on string-processing and is fully semantic. So, for example, if you want to categorise a class `A`, you simply define subclasses `A_cat1`, `A_cat2`, etc. corresponding to each of your categories. As the subclasses inherit all properties of the parent class, this is a really light-weight solution.